### PR TITLE
Add filedb package

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/datastore/filedb/filedb.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+type FileDB struct {
+	backend filestore.Store
+	logger  *zap.Logger
+}
+
+type Option func(*FileDB)
+
+func WithLogger(logger *zap.Logger) Option {
+	return func(m *FileDB) {
```
Comment (knanao) [inline pkg/datastore/filedb/filedb.go:33]: super nits👮 
```suggestion
	return func(f *FileDB) {
```

Comment (khanhtc1202) [reply pkg/datastore/filedb/filedb.go:33]: Nice catch 👍

File: pkg/datastore/filedb/filedb.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+type FileDB struct {
+	backend filestore.Store
+	logger  *zap.Logger
+}
+
+type Option func(*FileDB)
+
+func WithLogger(logger *zap.Logger) Option {
+	return func(f *FileDB) {
+		m.logger = logger
```
Comment (khanhtc1202) [inline pkg/datastore/filedb/filedb.go:34]:
```suggestion
		f.logger = logger
```

File: pkg/datastore/filedb/filedb.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+// limitations under the License.
+
+package filedb
+
+import (
+	"context"
+
+	"github.com/pipe-cd/pipecd/pkg/datastore"
+	"github.com/pipe-cd/pipecd/pkg/filestore"
+	"go.uber.org/zap"
```
Comment (nghialv) [inline pkg/datastore/filedb/filedb.go:22]: Move this to a separate import group.

**Review by nghialv — APPROVED** [review]

**Pull request merged**